renku/api/datasets.py

```diff
@@ -110,8 +110,53 @@ def with_dataset(self, name=None):
             #     if path.exists():
             #         raise ValueError('Dataset already exists')
 
-            with path.open('w') as f:
-                yaml.dump(source, f, default_flow_style=False)
+            if path is not None and path.parent.exists():
+                with path.open('w') as f:
+                    yaml.dump(source, f, default_flow_style=False)
+
+    def delete_dataset(self, dataset, force=False):
+        """Deletes empty dataset. If force flag is True,
+           it deletes all data from dataset.
+
+           raises: `DatasetNotEmpty` if dataset is not empty
+                    and --force flag is not used.
+        """
+        if not force and dataset.files:
+            raise DatasetNotEmpty()
+
+        dataset_dir_path = self.renku_datasets_path / dataset.identifier.hex
+        meta_file = dataset_dir_path / self.METADATA
+        meta_file.unlink()
+
+    def unlink_files(self, dataset, pattern, exclude=None):
+        """Unlinks files from dataset based on given pattern
+           and excludes those which match exclude pattern.
+
+           raises: ResourceNotFound if no matching dataset files are found
+        """
+        files_in_dataset = [(
+            key,
+            os.path.realpath(
+                self.renku_datasets_path / dataset.name / file.path
+            )
+        ) for key, file in dataset.files.items()]
```

Comment:
I am not sure if I was consistent everywhere but use `file_`, `filepath` or `path` instead of `file`.